codegen/gateway.go

```diff
@@ -777,6 +789,44 @@ func augmentHTTPEndpointSpec(
 	return espec, nil
 }
 
+func setPopulateMiddleware(middlewareObj map[string]interface{}) (map[string]FieldMapperEntry, error) {
+	fieldMap := make(map[string]FieldMapperEntry)
+	opts, ok := middlewareObj["options"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf(
+			"populate middleware found with no options.",
+		)
+	}
+	populates := opts["populate"].([]interface{})
+	for _, populate := range populates {
+		populateMap := populate.(map[string]interface{})
+		fromField, ok := populateMap["from"].(string)
+		if !ok {
+			return nil, errors.Errorf(
+				"populate middleware found with no source field.",
+			)
+		}
+		toField, ok := populateMap["to"].(string)
+		if !ok {
+			return nil, errors.Errorf(
+				"populate middleware found with no destination field.",
+			)
+		}
+		overrideOpt, ok := populateMap["override"].(bool)
+		if ok {
```

Comment:
This if else block is unnecessary.
```go
fieldMap[toField] = FieldMapperEntry{
        QualifiedName: fromField,
        Override: overrideOpt,
}
```
is sufficient.

Reply:
think your above 2 comments on same block:

I'll go with raising error, `override` has to be configured as bool, no fallback / error swallow -> no undefined behavior